sdk/rust-examples/minigrep/src/main.rs

```diff
@@ -0,0 +1,53 @@
+//! Mini Grep
+//!
+//! ## Context
+//!
+//! Searches for the argument inside the input file, and prints the result.
+//!
+//! ## Authors
+//!
+//! The Veracruz Development Team.
+//!
+//! ## Copyright
+//!
+//! See the file `LICENSING.markdown` in the Veracruz root directory for licensing and
+//! copyright information.
+
+use anyhow::{self, Ok};
+use std::{env, fs};
+
+const INPUT_FILENAME: &'static str = "/input/random_search_text.txt";
+const OUTPUT_FILENAME: &'static str = "/output/search_results.txt";
+
+fn main() -> anyhow::Result<()> {
+    let args: Vec<String> = env::args().collect();
+    let query: String = args.get(0).unwrap().to_string(); // Argument: To be searched for
+
+    let file_vec: Vec<u8> = fs::read(INPUT_FILENAME)?;
+
+    let filename: String = String::from_utf8_lossy(&file_vec).to_string(); // Input file: To be searched in
+
+    let res_vec: Vec<&str> = search(query.as_str(), &filename.as_str());
+
+    let mut ret: String = String::new();
+
+    for line in res_vec {
+        ret.push_str(line);
+    }
+
+    fs::write(OUTPUT_FILENAME, ret)?;
+
+    Ok(())
+}
+
+fn search<'a>(query: &str, contents: &'a str) -> Vec<&'a str> {
+    let mut results = Vec::new();
+
+    for line in contents.lines() {
```

Comment:
Is this really "grep" given you are searching for string literals, rather than matching patterns?  Maybe this can be generalised by finding a crate that actually implements regular expression matching, and using that (e.g., what does `ripgrep` use?) or changing the name to `search` rather than `minigrep`?

Reply:
I will change it to search then. But search is very generic. So can it be `string-search` instead?
(I am not using external crates currently, it'll take me a week or two, to master those syntaxes and rules)